### PR TITLE
RA-1093 added tests to openmrs.js

### PR DESCRIPTION
**Pull request by AdamGrzybkowski**

### CODE REVIEW

File: test/unit/openmrs.spec.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                        }
+                    }
+                }
+            }
+        });
+    });
+
+    beforeEach(module('conceptDictionaryApp'));
+
+    describe('openmrsRest', function(){
```
Comment (rkorytkowski) [inline test/unit/openmrs.spec.js]: describe should indicate what you are testing, e.g. "listFull should query and return full representations"

File: test/unit/openmrs.spec.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            $httpBackend.whenPOST('/ws/rest/v1/item').
+            respond({ name: 'item4', 
+                	uuid: "6g65dc56-72ec-4ab5-9fa8-d98be91adb5b"});
+            $httpBackend.whenPOST('/ws/rest/v1/item/6g65dc56-72ec-4ab5-9fa8-d98be91adb5b').
+            respond({ name: 'item5', 
+                	uuid: "6g65dc56-72ec-4ab5-9fa8-d98be91adb5b"});
+            $httpBackend.whenDELETE('/ws/rest/v1/item/6g65dc56-72ec-4ab5-9fa8-d98be91adb5b').
+            respond({message: 'Item deleted'});
+            
+            openmrsRest.listFull('item').then(function(resp){
```
Comment (rkorytkowski) [inline test/unit/openmrs.spec.js:40]: Please test one service method in one describe method.

File: test/unit/openmrs.spec.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+/* jasmine specs for controllers go here */
+describe('Concept dictionary controllers', function() {
```
Comment (rkorytkowski) [inline test/unit/openmrs.spec.js]: Please fix the description of this test suite.

File: test/unit/openmrs.spec.js (hunk 2 above)
Comment (AdamGrzybkowski) [inline test/unit/openmrs.spec.js:40]: Do you want me to split it into seperate files ? e.g. openmrsListFull.spec.js

Comment (rkorytkowski) [inline test/unit/openmrs.spec.js:40]: Not into separate files, but into separate describe methods.

File: test/unit/openmrs.spec.js [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    describe('Get should retrun one item', function(){
+        var $httpBackend, openmrsRest, item;
+
+        beforeEach(inject(function(_$httpBackend_, _openmrsRest_){
+        	openmrsRest = _openmrsRest_;
+            $httpBackend = _$httpBackend_;
+            $httpBackend.whenGET('partials/index-menu.html').respond();
+            $httpBackend.whenGET('/ws/rest/v1/item/6g65dc56-72ec-4ab5-9fa8-d98be91adb5b').
+            respond({ name: 'item3', 
+                	uuid: "6g65dc56-72ec-4ab5-9fa8-d98be91adb5b"});
```
Comment (rkorytkowski) [inline test/unit/openmrs.spec.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/openmrs/openmrs-owa-conceptdictionary/file/2319554381/issues/source?bid=3196969&fileBranchId=3204313#l66)

File: test/unit/openmrs.spec.js [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+            $httpBackend.flush();
+        }));
+        
+        it('should load items list', function(){
+        	expect(list.results).toEqualData([{ name: 'item1',  
+				            					uuid: "2b22dc27-72ec-4ab5-9fa8-d98be91adc1c"},
+				            				  { name: 'item2', 
+				            					uuid: "3g65dc27-72ec-4ab5-9fa8-d98be91adb5b"},
+				            					{ name: 'item3', 
+				                				uuid: "6g65dc56-72ec-4ab5-9fa8-d98be91adb5b"}]
```
Comment (rkorytkowski) [inline test/unit/openmrs.spec.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/openmrs/openmrs-owa-conceptdictionary/file/2319554381/issues/source?bid=3196969&fileBranchId=3204313#l53)

File: test/unit/openmrs.spec.js [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    describe('ListFull should retrun full representations', function(){
+        var $httpBackend, openmrsRest, list;
+
+        beforeEach(inject(function(_$httpBackend_, _openmrsRest_){
+        	openmrsRest = _openmrsRest_;
+            $httpBackend = _$httpBackend_;
+            $httpBackend.whenGET('partials/index-menu.html').respond();
+            $httpBackend.whenGET('/ws/rest/v1/item?v=full').
+            respond({results:[{ name: 'item1',  
+            					uuid: "2b22dc27-72ec-4ab5-9fa8-d98be91adc1c"},
```
Comment (rkorytkowski) [inline test/unit/openmrs.spec.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/openmrs/openmrs-owa-conceptdictionary/file/2319554381/issues/source?bid=3196969&fileBranchId=3204313#l33)

File: test/unit/openmrs.spec.js [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+
+        beforeEach(inject(function(_$httpBackend_, _openmrsRest_){
+        	openmrsRest = _openmrsRest_;
+            $httpBackend = _$httpBackend_;
+            $httpBackend.whenGET('partials/index-menu.html').respond();
+            $httpBackend.whenGET('/ws/rest/v1/item/6g65dc56-72ec-4ab5-9fa8-d98be91adb5b').
+            respond({ name: 'item3', 
+                	uuid: "6g65dc56-72ec-4ab5-9fa8-d98be91adb5b"});
+            
+            openmrsRest.get('item', {uuid: "6g65dc56-72ec-4ab5-9fa8-d98be91adb5b"}).then(function(resp){
```
Comment (rkorytkowski) [inline test/unit/openmrs.spec.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/openmrs/openmrs-owa-conceptdictionary/file/2319554381/issues/source?bid=3196969&fileBranchId=3204313#l68)

File: test/unit/openmrs.spec.js [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+
+        beforeEach(inject(function(_$httpBackend_, _openmrsRest_){
+	    	openmrsRest = _openmrsRest_;
+	        $httpBackend = _$httpBackend_;
+	        $httpBackend.whenGET('partials/index-menu.html').respond();
+	        $httpBackend.whenPOST('/ws/rest/v1/item').
+	        respond({ name: 'item4', 
+	            	uuid: "6g65dc56-72ec-4ab5-9fa8-d98be91adb5b"});
+	        
+	        var newItem = angular.toJson({name:'item4',uuid: "6g65dc56-72ec-4ab5-9fa8-d98be91adb5b"});
```
Comment (rkorytkowski) [inline test/unit/openmrs.spec.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/openmrs/openmrs-owa-conceptdictionary/file/2319554381/issues/source?bid=3196969&fileBranchId=3204313#l90)

File: test/unit/openmrs.spec.js [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+
+        beforeEach(inject(function(_$httpBackend_, _openmrsRest_){
+        	openmrsRest = _openmrsRest_;
+            $httpBackend = _$httpBackend_;
+            $httpBackend.whenGET('partials/index-menu.html').respond();
+            $httpBackend.whenGET('/ws/rest/v1/item?v=full').
+            respond({results:[{ name: 'item1',  
+            					uuid: "2b22dc27-72ec-4ab5-9fa8-d98be91adc1c"},
+            				  { name: 'item2', 
+            					uuid: "3g65dc27-72ec-4ab5-9fa8-d98be91adb5b"},
```
Comment (rkorytkowski) [inline test/unit/openmrs.spec.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/openmrs/openmrs-owa-conceptdictionary/file/2319554381/issues/source?bid=3196969&fileBranchId=3204313#l35)

File: test/unit/openmrs.spec.js [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+
+        beforeEach(inject(function(_$httpBackend_, _openmrsRest_){
+	    	openmrsRest = _openmrsRest_;
+	        $httpBackend = _$httpBackend_;
+	        $httpBackend.whenGET('partials/index-menu.html').respond();
+	        $httpBackend.whenPOST('/ws/rest/v1/item/6g65dc56-72ec-4ab5-9fa8-d98be91adb5b').
+	        respond({ name: 'item5', 
+	            	uuid: "6g65dc56-72ec-4ab5-9fa8-d98be91adb5b"});
+	        
+	        var updatedItem = angular.toJson({name:'item5',uuid: "6g65dc56-72ec-4ab5-9fa8-d98be91adb5b"})
```
Comment (rkorytkowski) [inline test/unit/openmrs.spec.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/openmrs/openmrs-owa-conceptdictionary/file/2319554381/issues/source?bid=3196969&fileBranchId=3204313#l113)

File: test/unit/openmrs.spec.js [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+	        openmrsRest.update('item',{uuid: "6g65dc56-72ec-4ab5-9fa8-d98be91adb5b"}, updatedItem).then(function(resp){
```
Comment (rkorytkowski) [inline test/unit/openmrs.spec.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/openmrs/openmrs-owa-conceptdictionary/file/2319554381/issues/source?bid=3196969&fileBranchId=3204313#l114)

File: test/unit/openmrs.spec.js [hunk 12]
```diff
@@ -0,0 +1,147 @@
...
+    	var $httpBackend, openmrsRest, DELETEitem;
+
+        beforeEach(inject(function(_$httpBackend_, _openmrsRest_){
+	    	openmrsRest = _openmrsRest_;
+	        $httpBackend = _$httpBackend_;
+	        $httpBackend.whenGET('partials/index-menu.html').respond();
+	        $httpBackend.whenDELETE('/ws/rest/v1/item/6g65dc56-72ec-4ab5-9fa8-d98be91adb5b').
+	        respond({message: 'Item deleted'});
+	        
+	        openmrsRest.remove('item',{uuid: "6g65dc56-72ec-4ab5-9fa8-d98be91adb5b"}).then(function(resp){
```
Comment (rkorytkowski) [inline test/unit/openmrs.spec.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/openmrs/openmrs-owa-conceptdictionary/file/2319554381/issues/source?bid=3196969&fileBranchId=3204313#l135)

File: test/unit/openmrs.spec.js [hunk 13]
```diff
@@ -0,0 +1,147 @@
...
+        	openmrsRest = _openmrsRest_;
+            $httpBackend = _$httpBackend_;
+            $httpBackend.whenGET('partials/index-menu.html').respond();
+            $httpBackend.whenGET('/ws/rest/v1/item?v=full').
+            respond({results:[{ name: 'item1',  
+            					uuid: "2b22dc27-72ec-4ab5-9fa8-d98be91adc1c"},
+            				  { name: 'item2', 
+            					uuid: "3g65dc27-72ec-4ab5-9fa8-d98be91adb5b"},
+            					{ name: 'item3', 
+                				uuid: "6g65dc56-72ec-4ab5-9fa8-d98be91adb5b"}
```
Comment (rkorytkowski) [inline test/unit/openmrs.spec.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/openmrs/openmrs-owa-conceptdictionary/file/2319554381/issues/source?bid=3196969&fileBranchId=3204313#l37)

**Comment by rkorytkowski** [comment]
Looks good. Please resolve conflicts and merge. Also read up on jasmine to better understand how to structure your tests http://jasmine.github.io/2.4/introduction.html

**Pull request merged**